addons/account/models/account_payment.py

```diff
@@ -659,6 +663,24 @@ def _compute_payment_receipt_title(self):
         """ To override in order to change the title displayed on the payment receipt report """
         self.payment_receipt_title = _('Payment Receipt')
 
+    @api.depends('partner_id', 'amount', 'date')
+    def _compute_duplicated_payment_ids(self):
+        """ Retrieve ids of payments with same partner_id, amount and date as the current payment """
+        for payment in self:
+            payment.has_duplicated_payments = 0
+            payment.duplicated_payment_ids = None
+            if payment.partner_id.id and payment.amount != 0:
+                domain = [('partner_id', '=', payment.partner_id.id),
```

Comment:
Weird indent on the domain, i would do something like 
![image](https://github.com/odoo/odoo/assets/109144554/590a1433-4745-4090-95f5-bee16f9212ac)